File: 8 - arquivo_texto/cSharp/Atomo.cs
public class Atomo 
{
    public string Sigla {get; set;}
    public string X {get; set;}
    public string Y {get; set;}
    public string Z {get; set;}

    public Atomo(string sigla, string x, string y, string z)
    {
        this.Sigla = sigla;
        this.X = x;
        this.Y = y;
        this.Z = z;
    }

    public override bool Equals(object? obj)
    {
        var atomo = obj as Atomo;

        if (atomo != null)
        { 
            return this.Sigla.Equals(atomo.Sigla) && this.X.Equals(atomo.X) &&
                    this.Y.Equals(atomo.Y) && this.Z.Equals(atomo.Z);

        } else
        {
            throw new ArgumentException("Objeto não é Atomo");
        }
    }

    public override string ToString()
    {
        return "Atomo{" + "sigla=" + this.Sigla + ", eixo x=" + this.X + ", eixo y=" + this.Y + ", eixo z=" + this.Z + '}';
    }

    public override int GetHashCode()
    {
        return HashCode.Combine(Sigla,X,Y,Z);
    }

}

File: 8 - arquivo_texto/java/Atomo.java
public class Atomo {
    String sigla;
    String x;
    String y;
    String z;
    
    public String getSigla() {
        return sigla;
    }
    public void setSigla(String sigla) {
        this.sigla = sigla;
    }
    public String getX() {
        return x;
    }
    public void setX(String x) {
        this.x = x;
    }
    public String getY() {
        return y;
    }
    public void setY(String y) {
        this.y = y;
    }
    public String getZ() {
        return z;
    }
    public void setZ(String z) {
        this.z = z;
    }
    public Atomo(String sigla, String x, String y, String z) {
        this.sigla = sigla;
        this.x = x;
        this.y = y;
        this.z = z;
    }
    @Override
    public int hashCode() {
        final int prime = 31;
        int result = 1;
        result = prime * result + ((sigla == null) ? 0 : sigla.hashCode());
        result = prime * result + ((x == null) ? 0 : x.hashCode());
        result = prime * result + ((y == null) ? 0 : y.hashCode());
        result = prime * result + ((z == null) ? 0 : z.hashCode());
        return result;
    }
    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (getClass() != obj.getClass())
            return false;
        Atomo other = (Atomo) obj;
        if (sigla == null) {
            if (other.sigla != null)
                return false;
        } else if (!sigla.equals(other.sigla))
            return false;
        if (x == null) {
            if (other.x != null)
                return false;
        } else if (!x.equals(other.x))
            return false;
        if (y == null) {
            if (other.y != null)
                return false;
        } else if (!y.equals(other.y))
            return false;
        if (z == null) {
            if (other.z != null)
                return false;
        } else if (!z.equals(other.z))
            return false;
        return true;
    }
    @Override
    public String toString() {
        return "Atomo [sigla=" + sigla + ", x=" + x + ", y=" + y + ", z=" + z + "]";
    }
}
